src/figuresClasses.ts

```diff
@@ -2,10 +1,87 @@
+type Color = 'red' | 'green' | 'blue';
+type Shape = 'triangle' | 'circle' | 'rectangle';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(
+        `The side of the figure cannot be less than or equal to zero. Invalid values: a = ${a}, b = ${b}, c = ${c}`,
+      );
+    }
+
+    const maxLength: number = Math.max(a, b, c);
+    const sumLength: number = a + b + c;
+
+    if (maxLength >= sumLength - maxLength) {
+      throw new Error(`Sides ${a}, ${b}, and ${c} can't form a triangle`);
+    }
```

Comment:
The condition correctly checks if the sides can form a triangle using the triangle inequality theorem. Ensure that the error message is clear and concise.